### ISSUE TITLE
Ubuntu 12.04.5 LTS: no import on boot with 0.6.4.x

### ISSUE THREAD

**Issue by MaFL2k** [zfsonlinux/zfs#3734]
Hi,

I have no more working auto import on boot with all releases of 0.6.4 so far.
<0.6.4 it was no problem.

Ubuntu 12.04.5 LTS

Things I tried so far:
- export/import pool, recreation of zpool.cache
- recreation of /etc/hostid
- spl.spl_hostid as kernel parameter (incorrect format prevented spl module loading, so it´s not ignored)
- echo manual >/etc/init/zpool-import.override

I´m running out of ideas ;)

Controller is LSI 9211-IT/mpt2sas, but all drives are online before the SPL/ZFS modules were loaded (and it worked flawlessly before 0.6.4)

SPL hostid message appears first on manual import:
[    9.063733] SPL: Loaded module v0.6.4.2-1~precise
[    9.087980] ZFS: Loaded module v0.6.4.2-1~precise, ZFS pool version 5000, ZFS filesystem version 5
[ 300.469277] SPL: using hostid 0x007f0101

Automount after import works.

kern.log: http://pastebin.com/Q5tFx6Lb

/var/log/upstart/zpool-import.log
/proc/self/fd/9: line 123: cannot create temp file for here-document: Read-only file system

root@vdr-mf:~# zpool status
  pool: bigpool
 state: ONLINE
  scan: resilvered 40K in 0h0m with 0 errors on Thu Sep  3 13:39:34 2015
config:

	NAME                                                 STATE     READ WRITE CKSUM
	bigpool                                              ONLINE       0     0     0
	  raidz1-0                                           ONLINE       0     0     0
	    scsi-SATA_ST4000VN000-1H4_W300QD4P               ONLINE       0     0     0
	    scsi-SATA_ST4000VN000-1H4_W300QENM               ONLINE       0     0     0
	    scsi-SATA_ST4000VN000-1H4_W300N1F5               ONLINE       0     0     0
	    scsi-SATA_ST4000VN000-1H4_W300QEN2               ONLINE       0     0     0
	  raidz1-1                                           ONLINE       0     0     0
	    scsi-SATA_ST4000VN000-1H4_S300RWB7               ONLINE       0     0     0
	    scsi-SATA_ST4000VN000-1H4_Z303SMBX               ONLINE       0     0     0
	    scsi-SATA_ST4000VN000-1H4_Z301APJ9               ONLINE       0     0     0
	    scsi-SATA_ST4000VN000-1H4_Z3028075               ONLINE       0     0     0
	  raidz1-2                                           ONLINE       0     0     0
	    scsi-SATA_ST2000VN000-1H3_W1H0XD3T               ONLINE       0     0     0
	    scsi-SATA_SAMSUNG_HD204UIS2HGJ1BB211257          ONLINE       0     0     0
	    scsi-SATA_SAMSUNG_HD204UIS2H7JD2B411416          ONLINE       0     0     0
	    scsi-SATA_SAMSUNG_HD204UIS2H7J9AB100870          ONLINE       0     0     0
	cache
	  scsi-SATA_INTEL_SSDSA2M04CVGB0181008L040NGN-part4  ONLINE       0     0     0

errors: No known data errors

Regards
Matt

**Comment by MaFL2k** [comment]
cat /var/log/upstart/zpool-import.log
/proc/self/fd/9: line 123: cannot create temp file for here-document: Read-only file system

**Comment by realG** [comment]
Same here, on a HP Microserver, with four SATA drives connected to the built in controller. They appear to be online before the ZFS/SPL modules are loaded.

kern.log: http://pastebin.com/SPz7wPJ2

I've skipped 0.6.4 and 0.6.4.1 so can't comment on those, but auto import was working for me with prior 0.6.3.x releases. 

```
$ dkms status
spl, 0.6.4.2, 3.2.0-89-generic, x86_64: installed
zfs, 0.6.4.2, 3.2.0-89-generic, x86_64: installed

$ uname -a
Linux haa 3.2.0-89-generic #127-Ubuntu SMP Tue Jul 28 09:28:15 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

I've tried the same troubleshooting steps as Matt above, no luck so far.

**Comment by sophana** [comment]
Hi

exact same problem for me after I updated my ubuntu precise server (in production!)
Had few troubles with zfs (needed uninstall reinstall)
Now, no pool is showing at boot.

it seems that zpool-import is not working at boot with the same error message in /var/log/upstart/zpool-import.log:

    /proc/self/fd/9: line 123: cannot create temp file for here-document: Read-only file system

when I start zpool-import manually it seems to work:
```
root@k1:~# zpool status
no pools available
root@k1:~# zpool import
   pool: vol
     id: 12047124859158282519
  state: ONLINE
 action: The pool can be imported using its name or numeric identifier.
 config:

        vol                                                 ONLINE
          mirror-0                                          ONLINE
            scsi-SATA_Hitachi_HUA7220_JPW9K0N13B6HML-part3  ONLINE
            scsi-SATA_Hitachi_HUA7220_JPW9K0N13JL7ZL-part3  ONLINE
root@k1:~# zpool status
no pools available
root@k1:~# initctl start zpool-import
zpool-import stop/waiting
root@k1:~# zpool status
  pool: vol
 state: ONLINE
  scan: scrub repaired 0 in 3h44m with 0 errors on Fri Oct 30 05:45:47 2015
config:

        NAME                                                STATE     READ WRITE CKSUM
        vol                                                 ONLINE       0     0     0
          mirror-0                                          ONLINE       0     0     0
            scsi-SATA_Hitachi_HUA7220_JPW9K0N13B6HML-part3  ONLINE       0     0     0
            scsi-SATA_Hitachi_HUA7220_JPW9K0N13JL7ZL-part3  ONLINE       0     0     0

errors: No known data errors
```

I don't know why this message about read-only file system. root fs is over ext4 over software raid1
My server is in production. Please help.

```
# dkms status
spl, 0.6.5.3, 3.2.0-93-generic, x86_64: installed
zfs, 0.6.5.3, 3.2.0-93-generic, x86_64: installed
# uname -a
Linux k1.worldspot.net 3.2.0-93-generic #133-Ubuntu SMP Fri Oct 23 13:32:16 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

Thanks

**Comment by sophana** [comment]
by googling, I found someone who could make some fixes to /etc/init/zpool-import.conf as a workaround.

https://github.com/zfsonlinux/zfs/issues/3033

However, his fixes don't work for me. I get no access to network when I workaround the read-only problem.
I found that during zpool-import.conf is executed, rootfs is readonly, and /run is a tmpfs filesystem.
As the server is remotely hosted, it was not easy to debug. (has to remote netboot into rescue everytime...)

The only solution I found was to replace the major part of the script by a simple 

    zpool import -a

I don't know the difference between this and the zpool-import script which is quite complicated. Maybe there is no timeout?

Someone has a better solution?
